File: Patterns/9. Diamond Patterns/86. Full Diamond Even Numbers/fden.cpp
#include <iostream>
#include <iomanip>
using namespace std;
int main(){
    int n,row;
    cout<<"Enter a Number : ";
    cin>>n;

    for(int i=1; i<=2*n-1; i++){
        if(i<=n){
            row = i;
        }else{
            row = 2*n-i;
        }
        // print spaces
        for(int j=1; j<=n-row; j++){
            cout<<"   ";
        }
        // print first half
        for(int j=1; j<=row; j++){
            cout<<setw(3)<<2*j;
        }
        // print second half
        for(int j=row-1; j>=1; j--){
            cout<<setw(3)<<2*j;
        }
        cout<<'\n';

    }
}

/*
Output

              2
           2  4  2
        2  4  6  4  2
     2  4  6  8  6  4  2
  2  4  6  8 10  8  6  4  2
     2  4  6  8  6  4  2
        2  4  6  4  2
           2  4  2
              2

*/

File: Patterns/9. Diamond Patterns/86. Full Diamond Even Numbers/fden.py
n = int(input("Enter a Number : "))
row=1

for i in range(1,2*n):
    if(i<=n):
        row=i
    else:
        row=2*n-i
    
    # print spaces
    for j in range(1,n-row+1):
        print(end='   ')
    # print first half
    for j in range(1,row+1):
        print(f"{2*j:3}", end='')
    # print second half
    for j in range(row-1,0,-1):
        print(f"{2*j:3}", end='')
    print()

'''
Output:

              2
           2  4  2
        2  4  6  4  2
     2  4  6  8  6  4  2
  2  4  6  8 10  8  6  4  2
     2  4  6  8  6  4  2
        2  4  6  4  2
           2  4  2
              2
         
'''